src/main/java/com/splunk/kafka/connect/SplunkSinkConnector.java

```diff
@@ -96,11 +100,81 @@ public Config validate(Map<String, String> connectorConfigs) {
                     ConfigValue::name,
                     Function.identity()));
 
-        validateAccess(connectorConfig, configValues);
+        HecConfig hecConfig = connectorConfig.getHecConfig();
+        try (CloseableHttpClient httpClient = createHttpClient(hecConfig)) {
+
+            if (!validateCollector(httpClient, hecConfig, configValues)) {
+                return config;
+            }
+
+            validateAccess(httpClient, hecConfig, configValues);
```

Comment:
Currently we are validating collector endpoint and access to these endpoints. Is there a way we could validate the hec token, i.e, when collector endpoint is valid and hec token is invalid could we record errors for token config?

Reply:
Yes, in the first method, the errors are recorded against the URI. And in the second method, since we will reach here only after the endpoint is validated, the errors are recorded against the token.